src/ai/backend/agent/docker/intrinsic.py

```diff
@@ -240,6 +241,51 @@ async def api_impl(container_id):
             ),
         ]
 
+    async def gather_process_measures(
+        self, ctx: StatContext, pids: Sequence[int]
+    ) -> Sequence[ProcessMeasurement]:
+        async def psutil_impl(pid: int):
+            try:
+                p = psutil.Process(pid)
+            except psutil.NoSuchProcess:
+                log.warning("psutil cannot found process {0}", pid)
+            else:
+                cpu_times = p.cpu_times()
+                cpu_used = Decimal(cpu_times.user + cpu_times.system) * 1000
+                return cpu_used
+            return None
```

Comment:
Using this kind of approach on macOS won't work. Containers will run on isolated VM aside from agent host.